src/ai/backend/manager/models/kernel.py

```diff
@@ -109,6 +109,42 @@ class KernelStatus(enum.StrEnum):
     ERROR = "ERROR"
     CANCELLED = "CANCELLED"
 
+    @classmethod
+    def from_str(cls, s: str) -> KernelStatus:
+        match s.upper():
+            case "PENDING":
+                return cls.PENDING
+            case "SCHEDULED":
+                return cls.SCHEDULED
+            case "PREPARING":
+                return cls.PREPARING
+            case "BUILDING":
+                return cls.BUILDING
+            case "PULLING":
+                return cls.PULLING
+            case "PREPARED":
+                return cls.PREPARED
+            case "CREATING":
+                return cls.CREATING
+            case "RUNNING":
+                return cls.RUNNING
+            case "RESTARTING":
+                return cls.RESTARTING
+            case "RESIZING":
+                return cls.RESIZING
+            case "SUSPENDED":
+                return cls.SUSPENDED
+            case "TERMINATING":
+                return cls.TERMINATING
+            case "TERMINATED":
+                return cls.TERMINATED
+            case "ERROR":
+                return cls.ERROR
+            case "CANCELLED":
+                return cls.CANCELLED
```

Comment:
Can you please fit the ones that are only applicable with upper (where name and value are not different) with cls(s.upper())?